web/src/main/java/org/springframework/security/web/server/authentication/AuthenticationWebFilter.java

```diff
@@ -95,11 +112,13 @@ public AuthenticationWebFilter(ReactiveAuthenticationManager authenticationManag
 	private Mono<Void> authenticate(ServerWebExchange exchange,
 		WebFilterChain chain, Authentication token) {
 		WebFilterExchange webFilterExchange = new WebFilterExchange(exchange, chain);
-		return this.authenticationManager.authenticate(token)
+
+		return this.authenticationManagerResolver.resolve(exchange.getRequest())
+			.map(authenticationManager -> authenticationManager.authenticate(token))
```

Comment:
@rhamedy I think what's necessary here is:

```suggestion
			.flatMap(authenticationManager -> authenticationManager.authenticate(token))
```

It works like `Optiona#flatMap` does, in that if a mapping method returns an `Optional`, then `flatMap` will strip it.

You should also be able to remove the cast at that point from line 119.

Reply:
That make sense! 👍